src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2619,18 +2659,23 @@ cldnn::network::ptr primitive_inst::get_unfused_subgraph() {
                         in = get_node().id();
                     }
                 }
+
+                if (has_primitive_id(outer_dep_ids_with_port, in.pid))
+                    in = tag_port_number(in.pid, in.idx);
+
+                GPU_DEBUG_TRACE_DETAIL << "    input of prim " << prim->id << "  - idx" << i << "  " << in << "\n";
             }
+            GPU_DEBUG_TRACE_DETAIL << "  add fused_primitive " << prim->id << "\n";
             t.add_primitive(prim);
             outer_dep_ids.push_back(prim->id);
         }
         // Samely, need to update dependency of the current fused nodes' input primitive ids with those in the current program
         for (size_t i = 0; i < prim_of_fused_node->input.size(); ++i) {
             auto& in = prim_of_fused_node->input[i];
-            if (std::find_if(outer_dep_ids.begin(), outer_dep_ids.end(),
-                             [&](const primitive_id& pid) {
-                                 return pid == in.pid;
-                             }) == outer_dep_ids.end()) {
-                in = get_node().get_dependency(i).id();
+            if (!has_primitive_id(outer_dep_ids, in.pid)) {
```

Comment:
Is this needed? It seems we updated all the dependencies at the very beginning

Reply:
It can be internal connection or external connection. As we are not sure, the name is not updated yet.

Reply:
Hmm, this logic updates the `prim_of_fused_node` dependencies - it shouldn't have any internal connections? As for external connections, they should have already been updated by this code?
```
            // Update input primitive name of prim_of_fused_node
            if (dep_idx < prim_of_fused_node->dependencies().size())
                prim_of_fused_node->get_dependency(dep_idx) = {port_dep_id, 0};
```

Reply:
oh sorry for my mis-explanation. You are right. I removed this code.

Reply:
Thank you!